### DISCUSSION TITLE
How to improve recording quality (from Apple Music to Logic Pro X via Blackhole)?

### DISCUSSION THREAD

**Discussion by Hiaubeng** [ExistentialAudio/BlackHole#732] [Q&A] [answered]
I has successfully used Blackhole driver to record an audio track that plays in Apple Music to Logic Pro X. However, the recorded audio quality is much worse than the original track. I have tinkered with Logic's recording parameters, but none seems to work. Is there anything I can do to improve the recording quality? Thank!

**Comment by Hiaubeng** [comment, marked as answer]
When the input device is set to be Blackhole instead of Multiple-output, the recording quality improves.

**Comment by devinroth** [comment]
What enters into BlackHole is what exits BlackHole. BlackHole does not change the quality. That means that the issue is how you have Logic setup. Or maybe your sample rate. But if I were to guess I would say you recorded in mono instead of stereo.

Devin

**Reply by Hiaubeng** [reply]
Thanks for your answer. So I adjust several parameters in Logic without really knowing what they do. Increasing sample rate in project setting from 41 to 48 kHz is one. And Pan Law is changed from -3dB to 0dB, which seems to increase recording track playback volume.  And in Logic Preferences, I/O Buffer Size is increased to maximum 1024. Taken together they improve the recording quality, but I don't know which parameter is the critical one.

**Reply by Hiaubeng** [reply]
In Channel Strip, I can choose stereo, mono, surround for output. If I choose stereo output, the playback sounds fuller, but in terms of clarity, mono output is still good.
